### DISCUSSION TITLE
[Development] Move source code into sub folder?

### DISCUSSION THREAD

**Discussion by stax76** [staxrip/staxrip#571] [Ideas]
@Dendraspis Maybe we should move the source code into a sub folder to get a nicer [main page](https://github.com/staxrip/staxrip).

Uncommon but I propose the name "Source".

I hope it would not cause lots of issues.

**Comment by Dendraspis** [comment]
Basically I have no problem with the current layout, but indeed it could make it easier to see and access `Changelog.md`.

And `Source` is definitely better than `src`.  😁

Currently I have no changes made, so feel free.  😊

**Reply by ghost** [reply]
> I hope it would not cause lots of issues.

I hope so, but is it right to assume that if it compiles successfully, then it means all parts are found, and there should be no issue at all ?

Out of the topic question: is Staxrip.exe in the beta's folder being released constantly after any commit ? (this is my impression currently)

**Reply by Dendraspis** [reply]
Seems you didn't want to reply to my answer.  😁

Right, and when he discovers issues he can always undo the changes before commiting.

**Reply by ghost** [reply]
> Seems you didn't want to reply to my answer. 😁

.... not me I guess 🤣 maybe you should put some @  😁

**Reply by stax76** [reply]
I move it now, since I'm not a git expert I hope it will work out well.

**Reply by stax76** [reply]
It's now committed, I hope it's fine. Maybe not pull it with VS but CLI or other git GUI.

**Reply by Dendraspis** [reply]
No problem as long as you can build it without issues.  😊

**Reply by ghost** [reply]
Looks better now indeed 😃

**Comment by Dendraspis** [comment]
The only downside it, that you have to switch to Folder View to open Changelog and Readme, but it's okay once they're open.  😊 
I'm curious when v2.1.9.0 Stable will be released ... or at least v.2.1.8.5.  😁 

BTW:  I refreshed the README.MD a bit.  😬

**Reply by ghost** [reply]
With major changes like dark theme and the highlights, it should be a 2.1.9.0 or even a 2.2 😁

**Reply by Dendraspis** [reply]
But first the Code Editor scrollbar flickering has to be fixed in either way.  🤔

**Reply by ghost** [reply]
@Dendraspis I agree, it can't remain like it is now...

Can you please  also check something: in the target file name text box, main window, when you modify the name of the target by typing something like "part one" inside the name, it lags more than it did before. I tried in in dark theme and in default theme, same thing.

It's minor, I know, but maybe there is an obvious reason that could be fixed...

**Reply by Dendraspis** [reply]
@44vince44 
Which version/hotfix/build do you use ?

**Reply by ghost** [reply]
last hotfix in the beta folder. i'm checking again that it wasn't replaced since then.

EDIT: it's still the same Staxrip.exe CRC32 E643C94E
EDIT2: YOUR BUILD Checkbox v6 CRC32 A988C8CE DOES NOT HAVE THIS ISSUE !!!

**Reply by stax76** [reply]
I've reverted it to your fix, couldn't get anything better.

Changlog.md is now included as a link.

Theming is a major improvement, so we can increase the major version, first and last time it was done by Revan when I was inactive.

**Reply by Dendraspis** [reply]
CRC32 is useless untill I know what was the last used commit.  😉 
But I can see what you mean - only happens with a (loaded) project, not using a template.

**Reply by stax76** [reply]
New hotfix is online.

**Reply by stax76** [reply]
The lag is maybe caused by synchronizing the script which happens now in the assistant. There is an exception, it's not synchronized when the resize slider is used but only on mouseup.

**Reply by ghost** [reply]
trying it now...

IT's WORSE !!!!! 😮 OMG

If you delete a few characters, you have to wait 6 seconds to see that you deleted them...

**Reply by stax76** [reply]
It also caused another old problem, a new script is written for every typed character.

**Reply by Dendraspis** [reply]
@stax76 
Didn't even know you can include files from another/upper directory.  😂 

But please have a look at the comment from @44vince44 above: https://github.com/staxrip/staxrip/discussions/571#discussioncomment-450145
When you have a project loaded and you change the target filename, it totally lags.
This must be related to one of you changes. I have checked my branch without your commits since yesterday and it works great - as @44vince44 also proved in this *EDIT2*.  

EDIT:  😂   Discussions are not refreshed when you already write a comment.  🙄

**Reply by stax76** [reply]
I've uploaded another hotfix.

I think again I did break a working change by Dendraspis..., sorry.

**Reply by Dendraspis** [reply]
@stax76 
Yeah, it must be the *Synchronize* thing.
BTW:  Same "issue" can be observed when you change the filename while the assistent is mocking about something.  Adding `-TEST1` to the filename results in 6 new script files.  But this is an old issue and not related to the new changes.

**Reply by Dendraspis** [reply]
At least you understand how hard it was to create ***Themes***.  😂

**Reply by ghost** [reply]
@stax76 the latest hotfix fixes the issue.
 Thanks !! 👍🏼 👍🏼

**Reply by ghost** [reply]
Now please @stax76 you should know that @Dendraspis has made several changes and has been very thorough in every case.

If you see something, it has maybe a reason.

For example, if you see a HACK where 0.1 is added to the bitrate of a certain audio encoder under some conditions, this is not an error, it is done on purpose, OK, so don't remove it...

🤣 
😁

**Reply by stax76** [reply]
I will try to be more careful and ask in doubt.
